src/ucs/config/parser.c

```diff
@@ -1625,13 +1618,45 @@ void ucs_config_parser_warn_unused_env_vars(const char *prefix)
         }
         ucs_warn("unused env variable%s:%s%s (set %s%s=n to suppress this warning)",
                  (num_unused_vars > 1) ? "s" : "", unused_env_vars_names,
-                 truncated ? "..." : "", UCS_CONFIG_PREFIX,
+                 truncated ? "..." : "", UCS_DEFAULT_ENV_PREFIX,
                  UCS_GLOBAL_OPTS_WARN_UNUSED_CONFIG);
     }
 
     pthread_mutex_unlock(&ucs_config_parser_env_vars_hash_lock);
 }
 
+void ucs_config_parser_warn_unused_env_vars_once(const char *env_prefix)
+{
+    int        base_prefix_added, env_prefix_added;
+    const char *base_prefix;
+
+    /* Although env_prefix is not real environment variable put it
+     * into table anyway to save prefixes which was already checked.
+     * Need to save both base_prefix and sub_prefix */
+    ucs_config_parser_mark_env_var_used(env_prefix, &env_prefix_added);
+    if (env_prefix_added) {
+        ucs_config_parser_warn_unused_env_vars_internal(env_prefix);
+        base_prefix = ucs_config_parser_get_base_prefix(env_prefix);
+
+        ucs_config_parser_mark_env_var_used(base_prefix, &base_prefix_added);
+        if (base_prefix_added) {
+            ucs_config_parser_warn_unused_env_vars_internal(base_prefix);
+        }
+    }
+}
+
+void ucs_config_parser_warn_unused_env_vars(const char *env_prefix)
```

Comment:
do we still need this function?
there is code duplication between this and ucs_config_parser_warn_unused_env_vars_once()

Reply:
ucs_config_parser_warn_unused_env_vars is not used, I just left it for the case if sometimes it will be useful to print warning messages multiple times. Is it ok to remove it?

Reply:
yes